pkg/utils/kube/kube.go

```diff
@@ -406,7 +406,7 @@ func GetDeploymentReplicas(u *unstructured.Unstructured) *int64 {
 
 // RetryUntilSucceed keep retrying given action with specified interval until action succeed or specified context is done.
 func RetryUntilSucceed(ctx context.Context, interval time.Duration, desc string, log logr.Logger, action func() error) {
-	pollErr := wait.PollImmediateUntil(interval, func() (bool /*done*/, error) {
+	pollErr := wait.PollUntilContextCancel(ctx, interval, true, func(_ context.Context) (bool /*done*/, error) {
```

Comment:
Just curious, why `PollUntilContextCancel` here instead of `PollUntilContextTimeout?`

Reply:
I would need a timeout and i didn't had one before.